src/jetstream.rs

```diff
@@ -20,7 +20,7 @@
 //! ```no_run
 //! # fn main() -> Result<(), Box<dyn std::error::Error>> {
 //! let nc = nats::connect("my_server::4222")?;
-//! let js = nats::jetstream::JetStream::new(nc, nats::jetstream::JetStreamOptions::default());
+//! let js = nats::jetstream::new(nc);
```

Comment:
Still prefer `let js = nc.jetstream(...)`

Reply:
Some rationale here, JetStream takes Connection by Arc (the type is an inner arc) and owns it
e.g, you can call `nats::jetstream::new(nats::connect(...)?);`.

Unlike Go, we don't need to keep a reference to the Connection for cleanup, that's all handled by drops. 
Having it as a factory doesn't add much other than a cyclical dependency imo.

That said, I'll give-in if either of you insist tho 🙂

Reply:
Well, my thumbs up was not about being able to clean things up, but having similar API to reference Client.

But I  have second thought:

We should keep API general design close to Go reference client (it has many pros), but retain Rust conventions.
I assume that Rust NATS users would appreciate API that feels familiar from Rust language perspective more than from other-language-client perspective.

If the API exposes similar functions (for subscribing, publishing, managing JetStream), has similar building blocks (like connection and JetStream context, JetStream management) that's already a big plus for any polyglot NATS user and for maintainers aiming for fast iterations on new features and features parity in general. But functions signatures, structs, patterns of their initialization and traits should follow Rust conventions.

That way  we have best of two worlds: we're able to deliver idiomatic, typical Rust code while retaining general feel of reference NATS client.